core/src/storage_stage.rs

```diff
@@ -454,6 +464,47 @@ impl StorageStage {
                             slot,
                             instruction_sender,
                         )?;
+                        // bundle up mining submissions from replicators
+                        // and submit them in a tx to the leader to get rewarded.
+                        let mut w_state = storage_state.write().unwrap();
+                        let instructions: Vec<_> = w_state
+                            .replicator_map
+                            .drain(..)
```

Comment:
does it empty the replicator_map?

Reply:
Yeah. I just realized we want that around. I'll fix it.

Reply:
On second thought, a drain is probably what we want(so we know which proofs to "validate"), but instead of draining the whole map, I'll just drain the proofs from it. That way we still know which replicator owns which segments

Reply:
Yea, that sounds good. I was thinking about where we use it for keeping track of which replicator has which segments.